components/scream/src/physics/shoc/shoc_calc_shoc_vertflux_impl.hpp

```diff
@@ -0,0 +1,40 @@
+#ifndef SHOC_CALC_SHOC_VERTFLUX_IMPL_HPP
+#define SHOC_CALC_SHOC_VERTFLUX_IMPL_HPP
+
+#include "shoc_functions.hpp" // for ETI only but harmless for GPU
+
+namespace scream {
+namespace shoc {
+
+template<typename S, typename D>
+void Functions<S,D>
+::calc_shoc_vertflux(
+  const MemberType& team,
+  const Int& shcol, const Int& nlev,
+  const uview_2d<const Spack>& tkh_zi,
+  const uview_2d<const Spack>& dz_zi,
+  const uview_2d<const Spack>& invar,
+  const uview_2d<Spack>& vertflux)
+{
+  const Int nlev_pack = scream::pack::npack<Spack>(nlev);
+  const auto sinvar = scalarize(invar);
+  Kokkos::parallel_for(Kokkos::TeamThreadRange(team, shcol), [&] (const Int& i) {
```

Comment:
Oh, wait, I'm looking at this function more closely. This actually isn't what we want. The top-level parallel_for, the kernel dispatch, will be over `i`, just like in P3. Functions of this sort will do a team || for, reduce, or scan over `k`. So there should be for `for (int k ...)` loop in this function, and the index passed to the ||4 should be `k`, not `i`.

Reply:
@ambrad , I was wondering about this. I looked up the callstack of calc_shoc_verflux and I landed inside a loop in shoc_main:

```
do t=1,nadv
```

which is over time steps. Not realizing that was over time steps, I initially assumed t would be the top-level parallel for. That obviously won't work since the time steps are dependent on each other. I will re-work this PR.